source/code/plugin/in_cadvisor_perf.rb

```diff
@@ -2,88 +2,89 @@
-  
-      def configure (conf)
-        super
+  class CAdvisor_Perf_Input < Input
+    Plugin.register_input("cadvisorperf", self)
+
+    def initialize
+      super
+      require "yaml"
+      require "json"
+
+      require_relative "CAdvisorMetricsAPIClient"
+      require_relative "oms_common"
+      require_relative "omslog"
+    end
+
+    config_param :run_interval, :time, :default => "1m"
+    config_param :tag, :string, :default => "oms.api.cadvisorperf"
+    config_param :healthtag, :string, :default => "oms.api.CIHealthPerf"
+
+    def configure(conf)
+      super
+    end
+
+    def start
+      if @run_interval
+        @finished = false
+        @condition = ConditionVariable.new
+        @mutex = Mutex.new
+        @thread = Thread.new(&method(:run_periodic))
       end
-  
-      def start
-        if @run_interval
-          @finished = false
-          @condition = ConditionVariable.new
-          @mutex = Mutex.new
-          @thread = Thread.new(&method(:run_periodic))
-        end
+    end
+
+    def shutdown
+      if @run_interval
+        @mutex.synchronize {
+          @finished = true
+          @condition.signal
+        }
+        @thread.join
       end
-  
-      def shutdown
-        if @run_interval
-          @mutex.synchronize {
-            @finished = true
-            @condition.signal
-          }
-          @thread.join
+    end
+
+    def enumerate()
+      time = Time.now.to_f
+      begin
+        eventStream = MultiEventStream.new
+        metricData = CAdvisorMetricsAPIClient.getMetrics()
+        metricData.each do |record|
+          record["DataType"] = "LINUX_PERF_BLOB"
+          record["IPName"] = "LogManagement"
+          eventStream.add(time, record) if record
+          #router.emit(@tag, time, record) if record
         end
-      end
-  
-      def enumerate()
-        time = Time.now.to_f
-        begin
-            eventStream = MultiEventStream.new
-            metricData = CAdvisorMetricsAPIClient.getMetrics()
-            metricData.each do |record|
-                    record['DataType'] = "LINUX_PERF_BLOB"
-                    record['IPName'] = "LogManagement"
-                    eventStream.add(time, record) if record
-                    #router.emit(@tag, time, record) if record    
-            end 
-            
-            router.emit_stream(@tag, eventStream) if eventStream
-            @@istestvar = ENV['ISTEST']
-            if (!@@istestvar.nil? && !@@istestvar.empty? && @@istestvar.casecmp('true') == 0 && eventStream.count > 0)
-              $log.info("cAdvisorPerfEmitStreamSuccess @ #{Time.now.utc.iso8601}")
-            end
-            rescue  => errorStr
-            $log.warn "Failed to retrieve cadvisor metric data: #{errorStr}"
-            $log.debug_backtrace(errorStr.backtrace)
+
+        router.emit_stream(@tag, eventStream) if eventStream
+        router.emit_stream(@healthtag, eventStream) if eventStream
```

Comment:
Changes in in_cadvisor_perf are mostly formatting, except for the part when a new event stream is emitted along with the existing one